pod/perlvar.pod

```diff
@@ -1828,19 +1828,40 @@ Also, it's just like C<$/>, but it's what you get "back" from Perl.
 X<$|> X<autoflush> X<flush> X<$OUTPUT_AUTOFLUSH>
 
 If set to nonzero, forces a flush right away and after every write or
-print on the currently selected output channel.  Default is 0
-(regardless of whether the channel is really buffered by the system or
-not; C<$|> tells you only whether you've asked Perl explicitly to
-flush after each write).  STDOUT will typically be line buffered if
-output is to the terminal and block buffered otherwise.  Setting this
-variable is useful primarily when you are outputting to a pipe or
-socket, such as when you are running a Perl program under B<rsh> and
-want to see the output as it's happening.  This has no effect on input
-buffering.  See L<perlfunc/getc> for that.  See L<perlfunc/select> on
-how to select the output channel.  See also L<IO::Handle>.
+print on the currently selected output channel. The default is 0.
+
+C<STDOUT> will typically be line buffered if output is to the terminal
+and block buffered otherwise. Setting this variable is useful
+primarily when you are outputting to a pipe or socket. This would
+come into play when you are writing to a pipe and want to see the
+output "live" with no buffering.
+
+B<Examples:>
+
+    $|      = 1;  # Set autoflush for current channel
+    $is_buf = $|; # Get the current autoflush status
+
+    STDOUT->autoflush(1); # Enable autoflush for STDOUT
+    STDERR->autoflush(0); # Disable autoflush for STDERR
+
+B<Warning:> C<$|> tells you only if you have asked Perl explicitly to
```

Comment:
This isn't exactly what the previous comment was saying. It is saying that the channel may be buffered or not, if it is not buffered already then autoflush won't have any effect but it still defaults to 0.

Reply:
This was worded weird originally which is why I tried to clean it up. I'm not sure what the correct less-wordy verbiage should be?

> `$|` tells you only if you have asked Perl explicitly to flush after each write. It is still possible the channel may be buffered by the system.

"**Warning:** External to Perl, the system may buffer a channel. If the system is *not* buffering a channel, then setting autoflush will not have any effect." ??

Reply:
I think rather than a warning, this statement is meant to be informational: the `$|`/autoflush setting always defaults to 0, and only indicates whether Perl flushes after each write, it does not indicate whether the channel is buffered by the system thus requiring flushing.

Reply:
Perhaps it would be better to have this information in the same place where it discusses the usual buffering of `STDOUT` since this is the property it's describing and not the autoflush behavior itself. And add that `STDERR` is typically unbuffered (but this may be system dependent, I am only familiar with Linux behavior).

Reply:
Also I was not being precise in saying autoflush won't have any effect on an unbuffered stream; Perl will still flush the handle after each write, it just won't make a difference because it wasn't being buffered anyway.

Reply:
The more I re-read this the more I think it _was_ just informational. Personally I didn't think about the system buffering a channel, so knowing that setting autoflush in Perl may **not** always do what you want seems appropriate to me.

I think my wording here is ok-ish?

Reply:
The current wording in the commit is not useful information. The only reason to turn on autoflush is *because* the channel may be buffered.